src/AppserverIo/Appserver/ServletEngine/Security/Auth/Spi/LdapLoginModule.php

```diff
@@ -0,0 +1,394 @@
+     * @param \AppserverIo\Collections\MapInterface                            $params          The parameters passed to the login module
+     *
+     * @return void
+     */
+    public function initialize(Subject $subject, CallbackHandlerInterface $callbackHandler, MapInterface $sharedState, MapInterface $params)
+    {
+
+        // call the parent method
+        parent::initialize($subject, $callbackHandler, $sharedState, $params);
+
+        // initialize the hash encoding to use
+        if ($params->exists(ParamKeys::URL)) {
+            $this->ldapUrl = $params->get(ParamKeys::URL);
+        }
+        if ($params->exists(ParamKeys::PORT)) {
+            $this->ldapPort = $params->get(ParamKeys::PORT);
+        }
+        if ($params->exists(ParamKeys::BASE_DN)) {
+            $this->baseDN = $params->get(ParamKeys::BASE_DN);
+        }
+        if ($params->exists(ParamKeys::BIND_DN)) {
+            $this->bindDN= $params->get(ParamKeys::BIND_DN);
+        }
+        if ($params->exists(ParamKeys::BIND_CREDENTIAL)) {
+            $this->bindCredential = $params->get(ParamKeys::BIND_CREDENTIAL);
+        }
+        if ($params->exists(ParamKeys::BASE_FILTER)) {
+            $this->baseFilter = $params->get(ParamKeys::BASE_FILTER);
+        }
+        if ($params->exists(ParamKeys::ROLES_DN)) {
+            $this->rolesDN = $params->get(ParamKeys::ROLES_DN);
+        }
+        if ($params->exists(ParamKeys::ROLE_FILTER)) {
+            $this->roleFilter = $params->get(ParamKeys::ROLE_FILTER);
+        }
+        if ($params->exists(ParamKeys::START_TLS)) {
+            $this->ldapStartTls = $params->get(ParamKeys::START_TLS);
+        }
+        if ($params->exists(ParamKeys::ALLOW_EMPTY_PASSWORDS)) {
+            $this->allowEmptyPasswords = $params->get(ParamKeys::ALLOW_EMPTY_PASSWORDS);
+        }
+        $this->setsMap = new HashMap();
+    }
+
+    /**
+     * Perform the authentication of username and password through LDAP.
+     *
+     * @return boolean TRUE when login has been successfull, else FALSE
+     * @throws \AppserverIo\Psr\Security\Auth\Login\LoginException Is thrown if an error during login occured
+     */
+    public function login()
+    {
+        $this->loginOk = false;
+
+        // array containing the username and password from the user's input
+        list ($name, $password) = $this->getUsernameAndPassword();
+
+        if ($name === null && $password === null) {
+            $this->identity = $this->unauthenticatedIdentity;
+        }
+
+        if ($this->identity === null) {
+            try {
+                $this->identity = $this->createIdentity($name);
+            } catch (\Exception $e) {
+                throw new LoginException(sprintf('Failed to create principal: %s', $e->getMessage()));
+            }
+        }
+        $ldap_connection = $this->ldapConnect();
```

Comment:
Refactor to 
```php
if ($ldapConnection = $this->ldapConnect()) {
...
}
```
and **ALWAYS** use camel case notation, means `$ldap_connection` should be `$ldapConnection`.